### PR TITLE
Add flatMap5...flatMap8 in Node java with corresponding Function5...F…

### PR DESCRIPTION
**Pull request by kshitij8**
…unction8 with unit tests

Ref: https://github.com/twitter/nodes/issues/9

### CODE REVIEW

File: src/main/java/com/twitter/nodes/Node.java [hunk 1]
```diff
@@ -1067,6 +1087,54 @@ private String getDebugPrefix() {
     };
   }
 
+  public static <T, A, B, C, D, E> Node<T> flatMap5(
+          String name,
```
Comment (ysaraf) [inline src/main/java/com/twitter/nodes/Node.java]: can you align indents with existing style?

Comment (kshitij8) [reply src/main/java/com/twitter/nodes/Node.java]: My bad, IntelliJ decided to be fancy.  
Updated the pull request.

**Comment by kshitij8** [comment]
I accidentally updated the branch request with `-f`, causing the comment to become outdated.  
Let me know if its a concern

File: src/main/java/com/twitter/nodes/Node.java [hunk 2]
```diff
@@ -1067,6 +1087,54 @@ private String getDebugPrefix() {
...
+  }
+
+  public static <T, A, B, C, D, E, F, G> Node<T> flatMap7(
+      String name,
+      Node<A> aNode, Node<B> bNode, Node<C> cNode, Node<D> dNode, Node<E> eNode, Node<F> fNode, Node<G> gNode,
+      Function7<Future<T>, A, B, C, D, E, F, G> func) {
+    return new Node<T>(name, aNode, bNode, cNode, dNode, eNode, fNode, gNode) {
+      @Override
+      protected Future<T> evaluate() throws Exception {
+        return func.apply(aNode.emit(), bNode.emit(), cNode.emit(), dNode.emit(), eNode.emit(), fNode.emit(), gNode.emit());
```
Comment (ysaraf) [inline src/main/java/com/twitter/nodes/Node.java]: some lines are too long now, please keep below 120 chars per line

File: src/main/java/com/twitter/nodes/Node.java [hunk 3]
```diff
@@ -1067,6 +1087,54 @@ private String getDebugPrefix() {
...
+      }
+    };
+  }
+
+  public static <T, A, B, C, D, E, F, G, H> Node<T> flatMap8(
+      String name,
+      Node<A> aNode, Node<B> bNode, Node<C> cNode, Node<D> dNode, Node<E> eNode, Node<F> fNode, Node<G> gNode, Node<H> hNode,
```
Comment (ysaraf) [inline src/main/java/com/twitter/nodes/Node.java]: same for this function also

File: src/main/java/com/twitter/nodes/Node.java (hunk 2 above)
Comment (kshitij8) [reply src/main/java/com/twitter/nodes/Node.java]: Done.

File: src/main/java/com/twitter/nodes/Node.java (hunk 3 above)
Comment (kshitij8) [reply src/main/java/com/twitter/nodes/Node.java]: Done

**Comment by kshitij8** [comment]
Did you get a chance to review it?
Any comments?

**Pull request merged**